Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/InstanceSynchronizationConfig.swift

```diff
@@ -90,15 +98,23 @@ internal struct InstanceSynchronization: Sequence {
             if var config = config.namespaces[namespace] {
                 return NamespaceSynchronization.init(namespacesColl: namespacesColl,
                                                      docsColl: docsColl,
-                                                     config: &config)
+                                                     config: &config,
+                                                     errorListener: errorListener)
             }
 
-            let newConfig = NamespaceSynchronization.init(namespacesColl: namespacesColl,
-                                                          docsColl: docsColl,
-                                                          namespace: namespace)
-            try! namespacesColl.insertOne(newConfig.config)
-            config.namespaces[namespace] = newConfig.config
-            return newConfig
+            do {
+                let newConfig = try NamespaceSynchronization.init(namespacesColl: namespacesColl,
+                                                                  docsColl: docsColl,
+                                                                  namespace: namespace,
+                                                                  errorListener: errorListener)
+                try namespacesColl.insertOne(newConfig.config)
+                config.namespaces[namespace] = newConfig.config
+                return newConfig
+            } catch {
+                errorListener?.on(error: error, forDocumentId: nil)
```

Comment:
[discussion] hmm I see how this can be weird. An error on a nil document ID probably means some fatal error has occurred. 

This would definitely not be in scope for this ticket, but we should maybe think about exposing a fatal error handler to differentiate this from normal sync errors that might freeze a document.

Also to answer your offline question about how we should handle errors that happen as part of CRUD operations. I think sync Android operations just throw, and we wrap that as the exception of the Task. For Swift, we'll probably just wrap anything that throws as the error in the `StitchResult<T>`.